libs/shared/src/utils/env.ts

```diff
@@ -0,0 +1,51 @@
+type CloudflareEnv = { env: Record<string, string> };
+
+// https://remix.run/blog/remix-vite-stable#cloudflare-pages-support
+const hasCloudflareProxyContext = (context: any): context is { cloudflare: CloudflareEnv } => {
+  return !!context?.cloudflare?.env;
+};
+
+const hasCloudflareContext = (context: any): context is CloudflareEnv => {
+  return !!context?.env;
+};
+
+/**
+ *
+ * Utility function to get env variables across Node and Edge runtimes.
+ *
+ * @param name Pass the name of the environment variable. The param is case-sensitive.
+ * @returns string Returns the value of the environment variable if exists.
+ */
+export const getEnvVariable = (name: string, context?: any): string => {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
**praise:** Really nice addition!‏